views/js/test/runner/provider/test.js

```diff
@@ -192,6 +192,44 @@ define([
             .render(container);
     });
 
+    QUnit.asyncTest('Clear a rendered item asynchronosuly', function(assert) {
+        var container = document.getElementById(containerId);
+
+        QUnit.expect(6);
+
+        assert.ok(container instanceof HTMLElement , 'the item container exists');
+        assert.equal(container.children.length, 0, 'the container has no children');
+
+        itemRunner.register('qti', qtiRuntimeProvider);
+
+        itemRunner('qti', itemData)
+            .on('render', function(){
+                assert.equal(typeof this._item, 'object', 'the item instance is attached to the runner');
+                assert.equal(container.children.length, 1, 'the container has children');
+
+                // Mock the getInteractions() method to return interaction with async clear step
+                this._item.getInteractions = function() {
+                    return [
+                        {
+                            clear: function() {
+                                return new Promise(function(resolve) { resolve(); });
```

Comment:
Could you please wrap `resolve` in a `setTimeout(resolve, 10)` so you simulate a real async call